### PR TITLE
dnsdist: Allow declaring custom metrics at runtime

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

Also fixes a bug in the prometheus HELP and TYPE messages for custom metrics with labels, and adds a method to increment a counter by more than one.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [x] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by rgacogne** [comment]
Added a new FFI method to declare a custom metric, and refactored the code to get rid of the duplication between Lua FFI/non-FFI.

**Review by wojas — APPROVED** [review]
I have tested the new metrics FFI interface inside a `LuaFFIPerThreadAction`. This worked as expected, allowing dynamic registration of additional metrics, with bulk increments. The metrics appeared with the right names in both the Prometheus `/metrics` and in the statistics API endpoint.

I have not looked in detail at the C++ code.

File: regression-tests.dnsdist/test_Advanced.py [hunk 1]
```diff
@@ -597,18 +597,19 @@ class TestCustomMetrics(DNSDistTest):
       initialCounter = getMetric("my-custom-counter")
       initialGauge = getMetric("my-custom-counter")
```
Comment (omoerbeek) [inline regression-tests.dnsdist/test_Advanced.py:598]:
```suggestion
      initialGauge = getMetric("my-custom-gauge")
```

File: pdns/dnsdistdist/docs/reference/custommetrics.rst [hunk 2]
```diff
@@ -1,9 +1,9 @@
 Custom Metrics
 =====================================
 
-You can define at configuration time your own metrics that can be updated using Lua.
+You can define at your own metrics that can be updated using Lua.
```
Comment (omoerbeek) [inline pdns/dnsdistdist/docs/reference/custommetrics.rst:4]:
```suggestion
You can define your own metrics that can be updated using Lua.
```

**Review by omoerbeek — APPROVED** [review]
Read through the code changes, luckily most are mechanical. Two nits.

**Comment by rgacogne** [comment]
Rebased on master to fix conflicts, plus fixed the two very good points raised by Otto!

**Pull request merged**